src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -123,6 +124,34 @@ sol_str_slice_to_string(const struct sol_str_slice slice)
     return strndup(slice.data, slice.len);
 }
 
+static inline struct sol_str_slice
+sol_str_slice_remove_leading_whitespace(struct sol_str_slice slice)
+{
+    char *data = (char *)slice.data;
+    size_t len = slice.len;
+
+    while (isspace(*data)) {
+        data++;
+        len--;
+    }
+
+    return SOL_STR_SLICE_STR(data, len);
+}
+
+static inline struct sol_str_slice
+sol_str_slice_remove_trailing_whitespace(struct sol_str_slice slice)
+{
+    char *ptr = (char *)slice.data + slice.len;
+    size_t len = slice.len;
+
+    while (ptr > (slice.data + slice.len - 1) && isspace(*ptr)) {
+        ptr--;
+        len--;
+    }
+
+    return SOL_STR_SLICE_STR(ptr, len);
+}
+
```

Comment:
yep, maybe, but I think it's not high priority and I'll postpone this to the end of my todo list (namely my linux-micro modules to be written).